### PR TITLE
Tweak: Nested Elements - Infra [ED-6591]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]
Please fix the checks errors

**Review by matipojo — CHANGES_REQUESTED** [review]
Please pull from 3.7

File: assets/dev/js/editor/components/settings/base/manager.js [hunk 1]
```diff
@@ -36,25 +36,28 @@ module.exports = elementorModules.ViewModule.extend( {
 		} );
 	},
 
-	getContainerId() {
```
Comment (matipojo) [inline assets/dev/js/editor/components/settings/base/manager.js:39]: BC

File: assets/dev/js/editor/document/repeater/commands/select.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			index: isRedo ? data.current : data.prev,
+		} );
+	}
+
+	validateArgs( args ) {
+		this.requireContainer( args );
+		this.requireArgumentType( 'index', 'number', args );
+
+		// When there multiple containers, then its not supported.
+		if ( args.containers ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/repeater/commands/select.js:20]: Why not?  We want to be prepared for multi edit.

File: assets/dev/js/editor/elements/manager.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			result = this.elementTypes[ key ];
+
+		if ( 'widget' === elType ) {
+			key = 'widget-' + widgetType;
+
+			if ( this.elementTypes[ key ] ) {
+				result = this.elementTypes[ key ];
+			}
+		}
+
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:37]: Please use `widgetType || elType`, in order to be prepared to unify the widget with the elements.

File: assets/dev/js/editor/elements/manager.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			this.registerElementType( element );
+		} );
+
+		// TODO: Find better place, since container is not module.
+		if ( elementorCommon.config.experimentalFeatures.container ) {
+			const ContainerClass = require( './types/container' ).default;
+
+			this.registerElementType( new ContainerClass() );
+		}
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:78]: Move to the container registration

File: assets/dev/js/editor/elements/manager.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import ElementBase from './types/base/element-base';
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:1]: Please cover with Jest tests

File: assets/dev/js/editor/elements/models/column.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import Element from './element';
```
Comment (matipojo) [inline assets/dev/js/editor/elements/models/column.js:1]: Please cover all models with Jest tests

File: assets/dev/js/editor/elements/types/base/element-base.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	/**
+	 * Function getTypeKey().
+	 *
+	 * Used to separate between elements and widgets.
+	 *
+	 * @return {String}
+	 */
+	getTypeKey() {
+		return 'element-' + this.getType();
+	}
```
Comment (matipojo) [inline assets/dev/js/editor/elements/types/base/element-base.js:39]: Remove

File: assets/dev/js/editor/elements/types/base/widget-base.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import ElementBase from './element-base';
+
+export default class WidgetBase extends ElementBase {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/types/base/widget-base.js:3]: Try to remove and use only the element base.

File: assets/dev/js/editor/elements/types/widget.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import Base from '../types/base/element-base';
+import Model from 'elementor-elements/models/widget';
+import { default as View } from 'elementor-elements/views/widget';
+
+export default class Widget extends Base {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/types/widget.js:5]: Try to remove and use the element base.

File: assets/dev/js/frontend/handlers/base-tabs-v2.js [hunk 10]
```diff
@@ -0,0 +1,223 @@
+export default class BaseTabsV2 extends elementorModules.frontend.handlers.Base {
```
Comment (matipojo) [inline assets/dev/js/frontend/handlers/base-tabs-v2.js:1]: Add the updated a11y to here

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/repeater/commands/select.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/repeater/commands/select.js:20]: How we can pass containers of two different tabs in the same time to select command?

File: assets/dev/js/editor/elements/manager.js (hunk 4 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js:78]: There is no place where container registered.

File: assets/dev/js/frontend/handlers/base-tabs-v2.js (hunk 10 above)
Comment (iNewLegend) [reply assets/dev/js/frontend/handlers/base-tabs-v2.js:1]: Cannot tabs-v2.scss does not exist here but in infra.

File: assets/dev/js/editor/elements/types/base/widget-base.js (hunk 8 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/types/base/widget-base.js:3]: I use it.

File: assets/dev/js/editor/components/settings/base/manager.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/components/settings/base/manager.js:39]: Handled https://github.com/elementor/elementor/pull/18297

File: assets/dev/js/editor/elements/manager.js (hunk 3 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js:37]: Handled https://github.com/elementor/elementor/pull/18297

File: assets/dev/js/editor/elements/manager.js (hunk 5 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js:1]: Handled https://github.com/elementor/elementor/pull/18297

File: assets/dev/js/editor/elements/models/column.js (hunk 6 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/models/column.js:1]: Handled https://github.com/elementor/elementor/pull/18297

File: assets/dev/js/editor/elements/types/base/element-base.js (hunk 7 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/types/base/element-base.js:39]: Handled https://github.com/elementor/elementor/pull/18297

File: assets/dev/js/editor/components/settings/page/manager.js [hunk 11]
```diff
@@ -63,7 +65,7 @@ module.exports = BaseSettings.extend( {
 		return this.editedView;
 	},
 
-	getContainerId() {
+	getContainerType() {
 		return 'document';
```
Comment (iNewLegend) [inline assets/dev/js/editor/components/settings/page/manager.js:69]: BC

File: assets/dev/js/editor/editor-base.js [hunk 12]
```diff
@@ -1077,6 +1087,8 @@ export default class EditorBase extends Marionette.Application {
 			this.generateResponsiveControlsForElements();
 		}
 
+		this.elementsManager = new ElementsManager();
+
 		this.initComponents();
```
Comment (matipojo) [inline assets/dev/js/editor/editor-base.js:1092]: Move to 
```suggestion
		this.initComponents();
```

File: assets/dev/js/editor/elements/manager.js [hunk 13]
```diff
@@ -17,25 +17,21 @@ export default class ElementsManager {
...
+	 * @returns {ElementBase}
 	 */
-	getElementType( elType, widgetType = false ) {
-		let key = 'element-' + elType,
-			result = this.elementTypes[ key ];
-
-		if ( 'widget' === elType ) {
-			key = 'widget-' + widgetType;
+	getElementType( key ) {
+		let type = this.elementTypes[ key ];
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:27]:
```suggestion
	getElementTypeClass( type ) {
		let typeClass = this.elementTypes[ type ];
```

File: assets/dev/js/editor/elements/manager.js [hunk 14]
```diff
@@ -49,13 +45,13 @@ export default class ElementsManager {
 			throw new TypeError( 'The element argument must be an instance of ElementBase.' );
 		}
 
-		const index = element.getTypeKey();
+		const key = element.getType();
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:48]:
```suggestion
		const type= element.getType();
```

File: assets/dev/js/editor/elements/manager.js [hunk 15]
```diff
@@ -49,13 +45,13 @@ export default class ElementsManager {
...
 
-		if ( this.elementTypes[ index ] ) {
+		if ( this.elementTypes[ key ] ) {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/manager.js:50]:
```suggestion
		if ( this.elementTypeClasses[ type ] ) {
```

File: assets/dev/js/editor/elements/views/base.js [hunk 16]
```diff
@@ -95,10 +96,11 @@ BaseElementView = BaseContainer.extend( {
 	},
 
 	getChildView( model ) {
-		const elementType = elementor.elementsManager.getElementType( model.get( 'elType' ), model.get( 'widgetType' ) );
+		const elementTypeKey = model.get( 'widgetType' ) || model.get( 'elType' ),
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/base.js:99]:
```suggestion
		const elementType = model.get( 'widgetType' ) || model.get( 'elType' ),
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/manager.js (hunk 14 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js:48]: Handled https://github.com/elementor/elementor/pull/18424

File: assets/dev/js/editor/editor-base.js (hunk 12 above)
Comment (iNewLegend) [reply assets/dev/js/editor/editor-base.js:1092]: Handled https://github.com/elementor/elementor/pull/18424

File: assets/dev/js/editor/elements/manager.js (hunk 13 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js:27]: Handled https://github.com/elementor/elementor/pull/18424

File: assets/dev/js/editor/elements/manager.js (hunk 15 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/manager.js:50]: Handled https://github.com/elementor/elementor/pull/18424

**Review by matipojo — CHANGES_REQUESTED** [review]
Please fix the checks errors

**Pull request merged**